examples/demo/src/reviews/ReviewCreate.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {
+    SimpleForm,
+    Create,
+    ReferenceInput,
+    TextInput,
+    DateInput,
+    AutocompleteInput,
+    required,
+    useNotify,
+    useRedirect,
+    getRecordFromLocation,
+} from 'react-admin';
+
+import StarRatingInput from './StarRatingInput';
+import { useLocation } from 'react-router';
+
+const ReviewCreate = () => {
+    const notify = useNotify();
+    const redirect = useRedirect();
+    const location = useLocation();
+
+    const onSuccess = (_: any) => {
+        const record = getRecordFromLocation(location);
+        notify('ra.notification.created');
+        if (record && record.product_id) {
+            redirect(`/products/${record.product_id}/reviews`);
+        } else {
+            redirect(`/reviews`);
+        }
+    };
+
+    return (
+        <Create mutationOptions={{ onSuccess }}>
+            <SimpleForm>
+                <ReferenceInput source="customer_id" reference="customers">
+                    <AutocompleteInput validate={required()} />
+                </ReferenceInput>
```

Comment:
One more thing: the created reviews miss a status. You can probably use `defaultValues` to set the default status to be `'pending'`.